### PR TITLE
#12673 Issue: On wide screens, the text in Learn > Classes is too wide to be comfortably readable

### PR DESCRIPTION
**Pull request by octopols**
Resolves: #12673

After struggling to compile MuseScore days I can finally compile it with no fatal errors and make changes and see them in the app itself (this is so cool!).

Tasks Done:
1. The card is now have a maximum size of 1000 pixels.
2. The card is now vertically scrollable (so many trials and errors. I may have made some unnecessary changes to make this work).
3. Divided the second paragraph into 2.
4. Removed the search box from

**I was not able to remove search box only from the classes tab in Learn, In this commit it's also removed from Get Started tab as well**

**I was not able to add extend the paragraph spacing to 6. ** 

- [x] I signed the [CLA](https://musescore.org/en/cla)
- [x] The title of the PR describes the problem it addresses
- [x] Each commit's message describes its purpose and effects, and references the issue it resolves
- [x] If changes are extensive, there is a sequence of easily reviewable commits
- [x] The code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] There are no unnecessary changes
- [x] The code compiles and runs on my machine, preferably after each commit individually
- [ ] I created a unit test or vtest to verify the changes I made (if applicable)

### CODE REVIEW

**Comment by octopols** [comment]
Okay so this commit fixes the search box problem. Now it's visible in `Learn > Get Started` and hidden in `Learn > Classes`

**Comment by jessjwilliamson** [comment]
Tested - works perfectly! Thank you!! @cbjeukendrup we happy to merge? :)

**Comment by Jojo-Schmitz** [comment]
Why closing it?

**Comment by octopols** [comment]
Sorry I'm reopening it now

File: src/learn/qml/MuseScore/Learn/LearnPage.qml [hunk 1]
```diff
@@ -120,6 +120,7 @@ FocusScope {
         }
     }
 
+
```
Comment (cbjeukendrup) [inline src/learn/qml/MuseScore/Learn/LearnPage.qml:123]: Unnecessary newline

File: src/learn/qml/MuseScore/Learn/LearnPage.qml [hunk 2]
```diff
@@ -167,6 +168,7 @@ FocusScope {
             navigation.name: "Get started"
             navigation.panel: navTabPanel
             navigation.column: 1
+            onClicked: searchField.visible = true
```
Comment (cbjeukendrup) [inline src/learn/qml/MuseScore/Learn/LearnPage.qml:171]: Code like this does not really fit into QML's declarative style. The stylistically "correct" way is to bind the visibility of the search field to the `currentIndex` of the `tabBar`. So, _inside the `SearchField { … }`_, you add:
```qml
visible: tabBar.currentIndex !== 3 // Not visible for Classes tab
```

File: src/learn/qml/MuseScore/Learn/internal/ClassesPage.qml [hunk 3]
```diff
@@ -63,141 +64,159 @@ FocusScope {
...
         anchors.right: parent.right
-        anchors.rightMargin: root.sideMargin
 
-        height: authorInfo.height
+        visible: root.contentHeight > root.height
+        z: 1
+    }
 
-        radius: 12
+    header: Item {
```
Comment (cbjeukendrup) [inline src/learn/qml/MuseScore/Learn/internal/ClassesPage.qml:78]: I might be missing something, but it looks like you're making it unnecessarily complicated. Why using a GridView with no items and only a header? Why not just `StyledFlickable`?

File: src/learn/qml/MuseScore/Learn/internal/ClassesPage.qml [hunk 4]
```diff
@@ -63,141 +64,159 @@ FocusScope {
...
+        Rectangle {
+            id: authorInfoContainer
+            anchors.top: parent.top
+            anchors.left: parent.left
+            anchors.leftMargin: root.sideMargin
+            anchors.rightMargin: root.sideMargin
 
-        color: ui.theme.backgroundPrimaryColor
+            height: authorInfo.height
+            width: (parent.width > 1100) ? 1000 : parent.width - 96
```
Comment (cbjeukendrup) [inline src/learn/qml/MuseScore/Learn/internal/ClassesPage.qml:90]: I'd replace `96` with `2 * root.sideMargin`, to make clear what it does.

Is there a reason for not writing `Math.min(1000, parent.width - root.sideMargin)`?

File: src/learn/qml/MuseScore/Learn/internal/ClassesPage.qml [hunk 5]
```diff
@@ -63,141 +64,159 @@ FocusScope {
...
 
-            height: childrenRect.height + 72
+            Column {
+                id: authorInfo
 
-            spacing: 30
+                anchors.left: parent.left
+                anchors.leftMargin: 36
+                anchors.rightMargin: 36
+                anchors.right: parent.right
```
Comment (cbjeukendrup) [inline src/learn/qml/MuseScore/Learn/internal/ClassesPage.qml:102]: This is a real nitpick, but if you're making amendments anyway, could you then please swap these two lines to make the order of the properties consistent for all three anchors? 
```suggestion
                anchors.right: parent.right
                anchors.rightMargin: 36
```

File: src/learn/qml/MuseScore/Learn/internal/ClassesPage.qml (hunk 3 above)
Comment (cbjeukendrup) [reply src/learn/qml/MuseScore/Learn/internal/ClassesPage.qml:78]: Also, if you change it to `StyledFlickable`, I think you won't need to put everything inside this `Item` anymore.

**Review by cbjeukendrup — COMMENTED** [review]
Hey, sorry for taking ages before I could review this. But here are some tips!

**Comment by cbjeukendrup** [comment]
Also, a rebase is needed. It looks like you've deleted your MuseScore fork at some point, so maybe you can't push to this PR anymore. If that's the case, feel free to close this one and create a new PR instead.

File: src/learn/qml/MuseScore/Learn/internal/ClassesPage.qml (hunk 3 above)
Comment (octopols) [reply src/learn/qml/MuseScore/Learn/internal/ClassesPage.qml:78]: I was completely new to QML when I opened this file. At that time, I simply relied on trial and error to explore what worked.
I will try to fix this today.

File: src/learn/qml/MuseScore/Learn/internal/ClassesPage.qml (hunk 5 above)
Comment (octopols) [reply src/learn/qml/MuseScore/Learn/internal/ClassesPage.qml:102]: I completely feel you here.

**Comment by octopols** [comment]
> Also, a rebase is needed. It looks like you've deleted your MuseScore fork at some point, so maybe you can't push to this PR anymore. If that's the case, feel free to close this one and create a new PR instead.

No, I no longer have access to this branch. I'll create a new PR.

**Pull request closed without merging**